tools/cldr-code/src/main/java/org/unicode/cldr/util/VoteResolver.java

```diff
@@ -262,7 +264,108 @@ public ImmutableSet<Integer> getVoteCountMenu() {
             admin.voteCountMenu = ImmutableSet.of(vetter.votes, admin.votes); /* Not LOCKING_VOTES; see canVoteWithCount */
             tc.voteCountMenu = ImmutableSet.of(vetter.votes, tc.votes, PERMANENT_VOTES);
         }
-    }
+
+        // The following methods were moved here from UserRegistry
+        // TODO: remove this todo notice
+
+        public boolean isAdmin() {
+            return stlevel <= admin.stlevel;
+        }
+
+        public boolean isTC() {
+            return stlevel <= tc.stlevel;
+        }
+
+        public boolean isExactlyManager() {
+            return stlevel == manager.stlevel;
+        }
+
+        public boolean isManagerOrStronger() {
+            return stlevel <= manager.stlevel;
+        }
+
+        public boolean isVetter() {
+            return stlevel <= vetter.stlevel;
+        }
+
+        public boolean isStreet() {
+            return stlevel <= street.stlevel;
+        }
+
+        public boolean isLocked() {
+            return stlevel == locked.stlevel;
+        }
+
+        public boolean isExactlyAnonymous() {
+            return stlevel == anonymous.stlevel;
+        }
+
+        /**
+         * Is this user an administrator 'over' this user? Always true if admin,
+         * or if TC in same org.
+         *
+         * @param org
+         */
+        public boolean isAdminForOrg(Organization myOrg, Organization target) {
+            boolean adminOrRelevantTc = isAdmin() ||
+
+                ((isTC() || stlevel == manager.stlevel) && (myOrg == target));
+            return adminOrRelevantTc;
+        }
+
+        public boolean canImportOldVotes(CheckCLDR.Phase inPhase) {
+            return isVetter() && (inPhase == Phase.SUBMISSION);
+        }
+
+        public boolean canDoList() {
+            return isVetter();
+        }
+
+        public boolean canCreateUsers() {
+            return isTC() || isExactlyManager();
+        }
+
+        public boolean canEmailUsers() {
+            return isTC() || isExactlyManager();
+        }
+
+        public boolean canModifyUsers() {
+            return isTC() || isExactlyManager();
+        }
+
+        public boolean canCreateOtherOrgs() {
+            return isAdmin();
+        }
+
+        public boolean canUseVettingSummary() {
+            return isManagerOrStronger();
+        }
+
+        public boolean canSubmit(CheckCLDR.Phase inPhase) {
+            if (inPhase == Phase.FINAL_TESTING) {
+                return false;
+                // TODO: Note, this will mean not just READONLY, but VETTING_CLOSED will return false here.
+                // This is probably desired!
+            }
+            return isStreet();
+        }
```

Comment:
where the old code uses SurveyMain.Phase.READONLY, the new code uses CheckCLDR.Phase.FINAL_TESTING, so this is potentially a change in functionality

@srl295 as you wrote in the commit message, "voting during VETTING_CLOSED should not have been allowed before, either, and probably stopped by other code" -- I imagine you're right; to be cautious I'll ask at today's infrastructure meeting, could there have been an intentional loophole for Admin or TC to vote during VETTING_CLOSED

Reply:
@btangmu thanks.

Reply:
i would lean towards making it more restrictive unless we have a reason this should be open.

Reply:
@srl295 at yesterday's infrastructure meeting it was expressed that ST may have VETTING_CLOSED during resolution, and that TC/Admin _need_ to be able to vote at that time

i'll study the question...

Reply:
locally, i've confirmed that TC can submit a new item during VETTING_CLOSED

Reply:
SurveyMain.Phase:
```
    public enum Phase {
        SUBMIT("Data Submission", CheckCLDR.Phase.SUBMISSION),
        VETTING("Vetting", CheckCLDR.Phase.VETTING),
        /*
         * VETTING_CLOSED means, "closed after vetting, while still open for admin tasks"
         */
        VETTING_CLOSED("Vetting Closed", CheckCLDR.Phase.FINAL_TESTING),
        CLOSED("Closed", CheckCLDR.Phase.FINAL_TESTING),
        DISPUTED("Dispute Resolution", CheckCLDR.Phase.VETTING),
        FINAL_TESTING("Final Testing", CheckCLDR.Phase.FINAL_TESTING),
        READONLY("Read-Only", CheckCLDR.Phase.FINAL_TESTING),
        BETA("Beta", CheckCLDR.Phase.SUBMISSION);
```

Reply:
CheckCldr.Phase:
```
    public enum Phase {
        BUILD, SUBMISSION, VETTING, FINAL_TESTING("RESOLUTION");

```

Reply:
it would be less confusing if there were only a single enum -- it could make all the fine distinctions while also providing some methods for treating some values as equivalent for some purposes

Reply:
Yup there's an issue for that. 

Maybe this one function shouldn't be refactored yet ?

Reply:
see my note above -- userCanSubmit (both versions), and canSubmit, are all essentially dead code if i'm not mistaken